lib/tarantool_server.py

```diff
@@ -920,6 +924,7 @@ def start(self, silent=True, wait=True, wait_load=True, rais=True, args=[],
                                                    actual_version),
                              schema='error')
                 raise TarantoolStartError(self.name)
+        return True
```

Comment:
What if we'll stuck in attempts to establish a connection in `wait_until_started()`? I think the whole `wait_until_started()` call should have one deadline, not just `wait_seek()`. Sketchy code to illustrate the idea:

```python
def start(<...>):
    deadline = time.time() + Options.args.server_start_timeout
    <..start tarantool..>
    wait_until_started(<...>, deadline)

def wait_until_started(<...>, deadline):
    wait_load(<...>, deadline)
    while time.time() < deadline:
        <..check connection..>

def wait_load(<...>, deadline):
    while time.time() < deadline:
        <..check logs..>
```

Reply:
I like this idea to prevent a connection stuck too, but in my opinion, `wait_load, wait_until_started` should take a timeout, not a deadline

Reply:
So, the real timeout will be `timeout * 2`: each step can take `timeout` seconds. If we'll have more steps in a future, it'll be `timeout * N`, and, what is worse, it'll not obvious from the code.

In fact, with current code, we have even more than `timeout * 2` at max: the first loop in `seek_wait()` has no check for a deadline and the deadline is calculated after it.

Reply:
You are right. It is unexpected behavior.

Reply:
I'll reopen this discussion to remind about the first loop in `seek_wait()`:

```python
while True:
    if os.path.exists(self.path):
        break
    gevent.sleep(0.001)
```

There is no deadline check here.

Reply:
It is still actual.

Reply:
Now fixed.